tests/ThreadParameters.lua

```diff
@@ -0,0 +1,73 @@
+-- Tests being able to send parameters to new threads on creation.
+
+local thread = require("thread");
+
+-- The amount of threads that will be created.
+local NUM_THREADS = 4;
+
+-- The maximum value to look for primes at.
+local CHECK_PRIMES_UNTIL = 400
+
+-- The amount of numbers each thread will have to check.
+local BATCH_SIZE_PER_THREAD = math.floor(CHECK_PRIMES_UNTIL / NUM_THREADS)
+
+
+-- Will contain all thread objects and their output.
+local threads = {}
+
+-- Checks if the provided number is a prime.
+-- Not an optimal implementation, but that's not the point of this test.
+local function isPrime(n)
+	if ((n == 2) or (n == 5)) then
+		-- 2 and 5 are primes.
+		return true;
+	elseif ((n % 2 == 0) or (n % 5 == 0)) then
+		-- Anything divisible by 2 or 5 is not a prime.
+		return false;
+	elseif (n == 1) then
+		-- 1 isn't a prime either.
+		return false;
+	end
+	
+    for i = 3, n / 2, 2 do
+        if (n % i) == 0 then
+            return false
+        end
+    end
+    return true
+end
+
+-- Thread function. Will check for primes in the provided limits.
+local thrFunc = function(outputIndex, min, max)
+	local output = {}
+	
+	-- Go through every number between min and max.
+	-- If the number is a prime save it in output.
+	for number = min, max do
+		if (isPrime(number)) then
+			table.insert(output, number);
+		end
+	end
+
+	-- Save the output of this thread in the threads variable.
+	threads[outputIndex].output = output;
+end
+
+-- Create new threads and provide the limits where they should look for primes as parameters.
+for I = 0, NUM_THREADS - 1 do
+	local idx = #threads + 1;
+	local batchStart = I * BATCH_SIZE_PER_THREAD;
+	local thr = thread.new(thrFunc, idx, batchStart, batchStart + BATCH_SIZE_PER_THREAD)
+	threads[idx] = {thread = thr, output = {}}
```

Comment:
Perhaps it's even better to create the output array outside the thread and send that as a parameter instead of the index.
```lua
-- Thread function. Will check for primes in the provided limits.
local thrFunc = function(output, min, max)
	-- Go through every number between min and max.
	-- If the number is a prime save it in output.
	for number = min, max do
		if (isPrime(number)) then
			table.insert(output, number);
		end
	end
end
```